Parse/src/main/java/com/parse/ParseACL.java

```diff
@@ -60,21 +61,63 @@ public static void setDefaultACL(ParseACL acl, boolean withAccessForCurrentUser)
    */
   //TODO (grantland): This should be a list for multiple lazy users with read/write permissions.
   private ParseUser unresolvedUser;
-  private JSONObject permissionsById;
+
+  /* package */ static class Permissions {
+    private static final String READ_PERMISSION = "read";
+    private static final String WRITE_PERMISSION = "write";
+
+    public boolean read, write;
+
+    /* package */ Permissions(boolean read, boolean write) {
+      this.read = read;
+      this.write = write;
+    }
+
+    /* package */ Permissions(Permissions permissions) {
+      this.read = permissions.read;
+      this.write = permissions.write;
+    }
+
+    /* package */ JSONObject toJSONObject() {
+      JSONObject json = new JSONObject();
+
+      try {
+        if (read) {
+          json.put(READ_PERMISSION, true);
+        }
+        if (write) {
+          json.put(WRITE_PERMISSION, true);
+        }
+      } catch (JSONException e) {
+        throw new RuntimeException(e);
+      }
+      return json;
+    }
+
+    /* package */ static Permissions createPermissionsFromJSONObject(JSONObject object) {
+      boolean read = object.optBoolean(READ_PERMISSION, false);
+      boolean write = object.optBoolean(WRITE_PERMISSION, false);
+      return new Permissions(read, write);
+    }
+
+    /* package */ boolean isEmpty() {
+      return !(read || write);
+    }
+  }
```

Comment:
Could we move this static class either above the instance variables or at the bottom of this class?